packages/cron/test/cargo.spec.js

```diff
@@ -0,0 +1,161 @@
+      [await randomCid(), 110, 110],
+      [await randomCid(), 125, 90], // To be updated
+      [await randomCid(), 0, 90], // To be updated
+      [await randomCid(), 100, null],
+      [await randomCid(), null, null],
+      [await randomCid(), 0, 0]
+    ]
+    for (let i = 0; i < contentItems.length; i++) {
+      const c = contentItems[i]
+      await Promise.all([
+        createUpload(dbClient, user._id, authKey, c[0], {
+          dagSize: c[1]
+        }),
+        createCargoDag(dbClient, {
+          cid_v1: c[0],
+          size_actual: c[2]
+        })
+      ])
+    }
+  })
+
+  it('updates the incorrect sizes', async () => {
+    await updateDagSizesWrp({ user })
+    const wrongSizes = contentItems.filter(getToBeUpdatedWrong)
+    await Promise.all(wrongSizes.map(async c => {
+      const u = await getUpload(dbClient, c[0], user._id)
+      assert.strictEqual(u.dagSize, c[2], `Not updated ${c[0]}`)
+    }))
+  })
+
+  it('updates the null sizes', async () => {
+    await updateDagSizesWrp({ user })
+    const nullSizes = contentItems.filter(getToBeUpdatedNull)
+    await Promise.all(nullSizes.map(async c => {
+      const u = await getUpload(dbClient, c[0], user._id)
+      assert.strictEqual(u.dagSize, c[2], `Not updated ${c[0]}`)
+    }))
+  })
+
+  it('returns the cids of all the updated content', async () => {
+    const { updatedCids } = await updateDagSizesWrp({ user })
+    const toBeUpdated = contentItems.filter((c) => getToBeUpdatedNull(c) || getToBeUpdatedWrong(c))
+    assert.strictEqual(toBeUpdated.length, updatedCids.length)
+    assert(toBeUpdated.map(e => e[0]).every((e) => updatedCids.includes(e)))
+  })
+
+  it('does not mess with correct ones', async () => {
+    const { allUploadsBefore, allUploadsAfter } = await updateDagSizesWrp({ user })
+    const correctSizes = contentItems.filter((c) => !getToBeUpdatedNull(c) && !getToBeUpdatedWrong(c))
+    await Promise.all(correctSizes.map(async c => {
+      const u = await getUpload(dbClient, c[0], user._id)
+      assert.strictEqual(u.dagSize, c[1])
+      assert.strictEqual(allUploadsBefore.find((u) => u.cid === c[0]).updated,
+        allUploadsAfter.find((u) => u.cid === c[0]).updated, `Updated date of ${c[0]}`)
+    }))
+  })
+
+  it('iterates until no more content needs updating', async () => {
+    // Limiting to one cid per iteration
+    const { updatedCids } = await updateDagSizesWrp({ user, limit: 1 })
+    const wrongSizes = contentItems.filter(getToBeUpdatedWrong)
+    await Promise.all(wrongSizes.map(async c => {
+      const u = await getUpload(dbClient, c[0], user._id)
+      assert.strictEqual(u.dagSize, c[2], `Not updated ${c[0]}`)
+    }))
+    const toBeUpdated = contentItems.filter((c) => getToBeUpdatedNull(c) || getToBeUpdatedWrong(c))
+
+    assert.strictEqual(toBeUpdated.length, updatedCids.length)
+  })
+
+  // Skipping this since for some reason the updated_at field  gets back as unchanged
```

Comment:
One or all of them? We need to resolve this - it sounds like a bug.

Reply:
This is actually passing now, got rid of the comment and `.skip`.